src/main/java/io/javalin/core/JavalinServer.kt

```diff
@@ -66,6 +67,9 @@ class JavalinServer(val config: JavalinConfig) {
             handler = attachJavalinHandlers(server.handler, HandlerList(httpHandler, webSocketHandler))
             connectors = connectors.takeIf { it.isNotEmpty() } ?: arrayOf(ServerConnector(server).apply {
                 this.port = serverPort
+                serverHost?.let {
```

Comment:
Could you set `serverHost` to be the default value, then do `this.host = serverHost` without the `let` ?

Reply:
Ok, I've done that with the commit below setting the default host to `"0.0.0.0"`.  

Yes, that makes the code nicer, yes I'm happy.  However, I note that strictly speaking the "default host" actually can take IPv6 into account:

In Jetty ServerConnector line 338 we see
```java
            InetSocketAddress bindAddress = getHost() == null ? new InetSocketAddress(getPort()) : new InetSocketAddress(getHost(), getPort());
```
So with host null Jetty uses `new InetSocketAddress(getPort())` ... and this internally uses `InetAddress.anyLocalAddress()` that I've put below.  The result of this for IPv6 might be different and for example be `::0`.

```java
    /*
     * Returns the InetAddress representing anyLocalAddress
     * (typically 0.0.0.0 or ::0)
     */
    static InetAddress anyLocalAddress() {
        return impl.anyLocalAddress();
    }
```

This is a long way of saying the default host somewhat depends on whether IPv4 or IPv6 is used / preferred.  So using `0.0.0.0` might not be the default people using IPv6 want or currently would get.

An alternative to just using "0.0.0.0" that would take IPv6 into account would be to use something like:

```java
    InetSocketAddress local = new InetSocketAddress(8090);
    final String hostName = local.getHostName();
```

Do people have comments / thoughts on this?

Reply:
FYI: I'm happy with the last commit that uses `InetSocketAddress(serverPort).hostName` as the default hostname.

Reply:
Thanks for being thorough @rbygrave. I think your solution is fine, but why did you create a method instead of setting the field directly?

I'm also wondering about the method for setting the host, maybe it should be an overload to start?

`app.start("0.0.0.0", 8090);` instead of `app.host("0.0.0.0").start(8090);`?

Reply:
> why did you create a method instead of setting the field directly?

No good reason - changing it :)

> method for setting the host, maybe it should be an overload to start?

Yes, all good with that.  

Pushed a commit with those changes.